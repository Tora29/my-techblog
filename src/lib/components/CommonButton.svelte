<script lang="ts">
	// コンポーネントの変数
	export let link: string = '/';
	export let classType: string = 'btn';
	export let text: string = 'Button';
	export let style: string = '';
	export let commonStyle: string = '';
</script>

<a href={link} class={`${classType} ${style} ${commonStyle}`}>{text}</a>
