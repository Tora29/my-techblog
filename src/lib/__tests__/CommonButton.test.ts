import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import CommonButton from '$lib/components/CommonButton.svelte';

describe('Button Component', () => {
	test('ボタンをデフォルトプロパティでレンダリング', () => {
		const { getByRole, getByText, container } = render(CommonButton);

		// リンクのデフォルト値を確認
		const link = getByRole('link');
		expect(link).toHaveAttribute('href', '/');

		// デフォルトのテキストを確認
		expect(getByText('Button')).toBeInTheDocument();

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});

	test('ボタンをカスタムプロパティでレンダリング', () => {
		const props = {
			link: 'test-link',
			text: 'test-text',
			classType: 'test-classType'
		};

		const { getByRole, getByText, container } = render(CommonButton, { props });

		// カスタムリンクを確認
		const link = getByRole('link');
		expect(link).toHaveAttribute('href', props.link);

		// カスタムテキストを確認
		expect(getByText(props.text)).toBeInTheDocument();

		// カスタムスタイルを確認
		expect(link).toHaveClass('btn', props.classType);

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});
});
