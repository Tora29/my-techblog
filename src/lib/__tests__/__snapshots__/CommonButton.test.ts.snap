// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > ボタンをカスタムプロパティでレンダリング 1`] = `
<div>
  <a
    class="btn test-classType "
    href="test-link"
  >
    test-text
  </a>
</div>
`;

exports[`Button Component > ボタンをデフォルトプロパティでレンダリング 1`] = `
<div>
  <a
    class="btn  "
    href="/"
  >
    Button
  </a>
</div>
`;
